### PR TITLE
Skip nearest target framework resolution if TargetFramework is specified and ReferringTargetFramework is empty

### LINKED ISSUE #416: The "GetNearestTargetFramework" task was not given a value for the required parameter "ReferringTargetFramework".
This is a scenario of Service Fabric application with an ASP.NET Core web service targeting .NET Framework.

In Visual Studio, Service Fabric Application project (similar to a Deployment project) adds project references to the service projects. With https://github.com/dotnet/sdk/pull/363, the SDK will check if referencing project and referenced project are compatible in terms of the target frameworks. However, Service Fabric Application project is not a managed code project, so it doesn't have any target framework moniker. In this case, the SDK reports an error on the referenced service project (ASP.NET Core on .NET FX)because of ReferringTargetFramework being empty.

As a tactical fix for RC.2, I suggest this:  If ReferringTargetFramework is empty, GetNearestTargetFramework returns the only TFM if the project only has one. Otherwise, it throws error. In this case, the semantic of ReferringTargetFramework being empty is that the referencing project can accept any TFM.

### PR DESCRIPTION
**Pull request by dsplaisted**
Fixes #416

### CODE REVIEW

File: src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.Common.targets [hunk 1]
```diff
@@ -59,13 +59,18 @@ Copyright (c) .NET Foundation. All rights reserved.
   <Target Name="GetTargetFrameworkProperties" Returns="TargetFramework=$(NearestTargetFramework)">
 
     <PropertyGroup>
+      <!-- If a ReferringTargetFramework was not specified, and we only have one TargetFramework, then don't try to check compatibility -->
+      <_SkipNearestTargetFrameworkResolution Condition="'$(TargetFramework)' != '' and '$(ReferringTargetFramework)' == ''">true</_SkipNearestTargetFrameworkResolution>
+      <NearestTargetFramework Condition="'$(_SkipNearestTargetFrameworkResolution)' != 'true'">$(TargetFramework)</NearestTargetFramework>
```
Comment (weidazhao) [inline src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.Common.targets]: Should it be ==?

**Review by nguerrera — APPROVED** [review]
I wonder if we should just pick the first one for x-targeting case. The folks with the deployment project are saying they can't provide a specific referring framework, so I think that's the best we can do. Unless it's acceptable that these deployment projects just don't work at all with x-targeted apps?

**Comment by dsplaisted** [comment]
I've filed Microsoft/MSBuild#1390 to provide a hook besides `TargetFrameworkMoniker` to allow the `ReferringTargetFramework` passed to project references to be provided.  If the deployment project doesn't know specifically which version of .NET it will be deployed to, then it can at least specify the highest possible .NET version, so that the referenced project would build for .NET Framework instead of .NET Core, regardless of which was listed first in the `TargetFrameworks` property.

**Comment by fleed** [comment]
I'm experiencing issues when trying to add to a ServiceFabric application a project that uses both `net461` and `netcoreapp1.1` as target frameworks. Is it related to this issue? Is there a workaround?
I'm using `dotnet` version `1.0.4`. The specific error is:

```
error MSB4044: The "GetNearestTargetFramework" task was not given a value for the required parameter "ReferringTargetFramework"
```

**Pull request merged**